modules/test/playwright/tests/client-extension-web/themeCSS.spec.ts

```diff
@@ -0,0 +1,211 @@
+
+	const fileChooser = await fileChooserPromise;
+
+	await fileChooser.setFiles(
+		path.join(__dirname, '/dependencies/' + fileName)
+	);
+
+	await expect(page.getByText(message)).toBeVisible();
+};
+
+test('ThemeCSS Client Extension supports Frontend Token Definition file upload', async ({
+	apiHelpers,
+	clientExtensionsPage,
+	page,
+}) => {
+	await apiHelpers.featureFlag.updateFeatureFlag('LPD-10773', true);
+
+	await clientExtensionsPage.goto();
+
+	await clientExtensionsPage.newClientExtensionButton.click();
+
+	await clientExtensionsPage.addThemeCSSMenuItem.click();
+
+	await uploadAndValidateFile(
+		clientExtensionsPage,
+		'invalid-frontend-token-definition.json',
+		'The format is not valid. Please upload a valid Frontend Token Definition JSON file.',
+		page
+	);
+
+	await uploadAndValidateFile(
+		clientExtensionsPage,
+		'empty-frontend-token-definition.json',
+		'Frontend Token Definition uploaded. Contributing 0 Token Categories, 0 Token Sets, and 0 Tokens.',
+		page
+	);
+
+	await uploadAndValidateFile(
+		clientExtensionsPage,
+		'empty-json-file.json',
+		'Frontend Token Definition uploaded. Contributing 0 Token Categories, 0 Token Sets, and 0 Tokens.',
+		page
+	);
+
+	await uploadAndValidateFile(
+		clientExtensionsPage,
+		'frontend-token-definition.json',
+		'Frontend Token Definition uploaded. Contributing 1 Token Categories, 1 Token Sets, and 2 Tokens.',
+		page
+	);
+});
+
+test('ThemeCSS Client Extension Frontend Token Definition tokens appears on new stylebook', async ({
+	apiHelpers,
+	clientExtensionsPage,
+	knowledgeBasePage: {productMenuPage},
+	page,
+}) => {
+	await apiHelpers.featureFlag.updateFeatureFlag('LPD-10773', true);
+
+	// Create Theme CSS Client Extension
+
+	await clientExtensionsPage.goto();
+
+	await clientExtensionsPage.newClientExtensionButton.click();
+
+	await clientExtensionsPage.addThemeCSSMenuItem.click();
+
+	const clientExtensionName = getRandomId();
+
+	await page
+		.locator(
+			'#_com_liferay_client_extension_web_internal_portlet_ClientExtensionAdminPortlet_name'
+		)
+		.fill(clientExtensionName);
+
+	await uploadAndValidateFile(
+		clientExtensionsPage,
+		'frontend-token-definition.json',
+		'Frontend Token Definition uploaded. Contributing 1 Token Categories, 1 Token Sets, and 2 Tokens.',
+		page
+	);
+
+	// Save Client Extension
+
+	await page.click(
+		'#_com_liferay_client_extension_web_internal_portlet_ClientExtensionAdminPortlet_editClientExtensionEntrySubmitButton'
+	);
+
+	// Navigate to Pages in Site Builder
+
+	await productMenuPage.goToPagesMenuItem();
+
+	await page
+		.locator(
+			`//*[@id="_com_liferay_layout_admin_web_portlet_GroupPagesPortlet_controlMenu"]/div[4]/ul/li[1]/div/div[1]/button`
+		)
+		.click();
+
+	await page
+		.locator('.dropdown-item')
+		.and(page.getByText('Configuration'))
+		.click();
+
+	await page
+		.locator(
+			'#_com_liferay_layout_admin_web_portlet_GroupPagesPortlet_themeCSSReplacementExtension'
```

Comment:
This stuff looks like it should probably be in a new global PagesAdminPage

Reply:
Which lines range are you talking about? Or is that only for the 138?